__tests__/Unit/pages/Dashboard/dashboardTest.test.tsx

```diff
@@ -0,0 +1,56 @@
+import DashboardPage from '@/pages/dashboard';
+import { renderWithRouter } from '@/test_utils/createMockRouter';
+import { Provider } from 'react-redux';
+import { store } from '@/app/store';
+import { fireEvent, screen } from '@testing-library/react';
+
+describe('dashboard page test', function () {
+    it('checks if the page is rendered with exact components', function () {
+        renderWithRouter(
+            <Provider store={store()}>
+                <DashboardPage />
+            </Provider>,
+            { query: { dev: 'true' } }
+        );
+
+        const searchBar = screen.getByTestId('searchbar_input');
+        expect(searchBar).toBeInTheDocument();
+
+        const searchButton = screen.getByTestId('search_btn');
+        expect(searchButton.innerHTML).toBe('Search');
+    });
+
+    it('renders 404 without passing the feature flag', function () {
+        renderWithRouter(
+            <Provider store={store()}>
+                <DashboardPage />
+            </Provider>
+        );
+
+        const headings = screen.getAllByRole('heading');
+        expect(headings).toHaveLength(1);
+        expect(headings[0].innerHTML).toBe('404 - Page Not Found');
```

Comment:
If there was only one heading, why use `getAllByRole`?

Reply:
To verify the same thing that there is only 1 heading as you can see there is an expect that checks only 1 heading is present. in case we find more I want the test to fail, using getByRole will return te first found heading in case if in future someone plans to add 1 more heading the changed tests (changes suggested in the comment above) wont be able to catch that and still pass assuming the first heading caught had a 404 text.

Reply:
Got it.